### DISCUSSION TITLE
realityObject的dest能否设为一个IP:PORT并设定SNI或Host呢

### DISCUSSION THREAD

**Discussion by caikun233** [XTLS/Xray-core#4094] [General]
就是说让Reality使用我指定的SNI（或者Host）访问dest的地址。
背景需求有点少见了，但大抵是这样的：

1. 我希望Reality获取证书的网站，跟Reality所在的服务器处于同一个局域网
2. 希望Reality通过局域网建立与dest服务器的连接，而不是公网。这样如果有人尝试通过Reality刷我流量的话，局域网通讯不要钱所以能省几个子
3. 但是dest服务器的DNS解析又全是公网
4. 没有Administrators，没办法写host

帮朋友问的，太抽象了，需求刁钻闻所未闻，拼尽全力无法战胜，故start a new discussion.

**Comment by Waldo-Schaeffer** [comment]
我也有类似的架构，主要是伪装成CDN服务器，但是我用的是自己的域名：
前置nginx的stream块，监听443，SNI分流：
1、self走reality(127.0.0.18:4444)
2、其余SNI白名单走反向代理4445
3、不在白名单中的SNI走4446
reality的dest填写127.0.0.1:4445，域名写self
server块，监听127.0.0.1:4445，放好所有的网站和证书
stream块，监听127.0.0.1:4446，反向代理cdn（比如B站海外的直播CDN）

**Comment by ZqinKing** [comment]
可以写IP:PORT,在reality后面放一个lucky简直完美。

**Reply by Waldo-Schaeffer** [reply]
我指的是ip:port，但是还要指定SNI，或者HOST

**Reply by ZqinKing** [reply]
> 我指的是ip:port，但是还要指定SNI，或者HOST

入站的reality不需要指定SNI。浏览器访问reality监听端口的时候会携带SNI，URL里写啥SNI就是啥。都会fallback到dest去。
